lib/main.py

```diff
@@ -27,20 +29,46 @@ async def lifespan(_app: FastAPI):
 
 
 APP = FastAPI(lifespan=lifespan)
-TASK_LIST: queue.Queue = queue.Queue(maxsize=100)
 
 
 class BackgroundProcessTask(threading.Thread):
     def run(self, *args, **kwargs):  # pylint: disable=unused-argument
+        nc = NextcloudApp()
+
+        task_type_ids = set()
+        for chain_name, _ in chains.items():
+            (model, task) = chain_name.split(":", 2)
+            task_type_ids.add("core:text2text:" + task)
+
         while True:
-            task = TASK_LIST.get(block=True)
+            # Reset user
+            nc.set_user("")
+            response = nc.providers.task_processing.next_task(list(task_type_ids))
+            if not isinstance(response, dict):
+                time.sleep(5)
+                continue
+
+            task = response["task"]
+            provider = response["provider"]
+            print(task)
+            print(provider)
+
+            nc.set_user(task["userId"])
```

Comment:
Whole llm2 should work without any `set_user`. 
Implementation of `setResult` API in Server repo should be changed, mostly the `getUserTask` function and call to it - for ExApps it should be allowed to set result of any task without impersonating user.

Reply:
Yeah I saw this as well and I don't know why @marcelklehr implemented it this way, but I also don't see a problem doing it this way as we get the user from the task and can simply impersonate them without a problem.